packages/fcl/src/current-user/index.js

```diff
@@ -93,44 +93,62 @@ function notExpired(user) {
   )
 }
 
+async function makeMessage() {
+  return {
+    timestamp: Date.now(),
+    appDomainTag: await config.get("fcl.appDomainTag"),
+    extensions: window.fcl_extensions || [],
+  }
+}
+
 async function authenticate({service, redir = false}) {
   return new Promise(async (resolve, reject) => {
     spawnCurrentUser()
     const user = await snapshot()
-    if (user.loggedIn) return resolve(user)
-
-    const {discoveryWallet, discoveryWalletMethod, appDomainTag} =
-      await buildAuthnConfig()
+    const msg = await makeMessage()
+    const opts = {redir}
+    const discoveryService = await getDiscoveryService()
+    const refreshService = serviceOfType(user.services, "authn-refresh")
+    const suppressRedirWarning = await config.get("fcl.warning.suppress.redir")
 
+    if (redir && !suppressRedirWarning) {
+      console.warn(
+        `You are manually enabling a very experimental feature that is not yet standard, use at your own risk.
+         You can disable this warning by setting fcl.warning.suppress.redir to true in your config`
+      )
+    }
     invariant(
-      service || discoveryWallet,
+      service || discoveryService.endpoint,
       `
         If no service passed to "authenticate," then "discovery.wallet" must be defined in config.
         See: "https://docs.onflow.org/fcl/reference/api/#setting-configuration-values"
       `
     )
 
-    const suppressRedirWarning = await config.get("fcl.warning.suppress.redir")
-    if (redir && !suppressRedirWarning) {
-      console.warn(
-        `You are manually enabling a very experimental feature that is not yet standard, use at your own risk.
-         You can disable this warning by setting fcl.warning.suppress.redir to true in your config`
-      )
+    if (user.loggedIn) {
+      if (refreshService) {
+        try {
+          const response = await execService({
+            service: refreshService,
+            msg,
+            opts,
+          })
+          send(NAME, SET_CURRENT_USER, await buildUser(response))
+        } catch (e) {
+          console.error("Error Refreshing Session. Session may be expired", e)
```

Comment:
Isn't the job of the authn refresh service to prevent this error? Perhaps the error message should say something like: "Error: Could not refreshing authentication."

Reply:
Yeah, I think that's a better error message